lib/context_functions_generator.ex

```diff
@@ -105,8 +105,15 @@ defmodule ContextFunctionsGenerator do
 
   def generate_function(:count, _name, pluralized_name, module, opts) do
     quote do
-      def unquote(:"count_#{pluralized_name}")(field \\ :id, repo_opts \\ []) do
+      def unquote(:"count_#{pluralized_name}")(opts \\ [], repo_opts \\ []) do
```

Comment:
2 things here:

1. What do you think about adding typespecs?
2. This is a breaking change, since the first parameter is changing. So there's two options here, accept the breaking change and start planning the next major version (since there are plans to also change the `list_` functions as well) or we can add a guard `when is_atom(field)` and `when is_map(count_params) or when is_list(count_params)`, keeping the compatibility

Reply:
cc @gabrielpra1

Reply:
I think we can create a more specific problem for using typespecs since it's something we're not using yet, taking advantage of the idea of ​​doing this for filters and other functions.

About the rupture change, we can make this adjustment using guard  :+1: 
Until we created a new version removing all deprecated functions.

Reply:
1. Okay, probably we could add typespecs later.

2. Agreed!